### ISSUE TITLE
RangeError: Maximum call stack size exceeded

### ISSUE THREAD

**Issue by bday2009** [Quramy/ts-graphql-plugin#290]
First and foremost, thank you to @Quramy for this incredible plugin, the types it creates are very easy to be consumed in a React project. Recently I ran into some problem when it failed to generate types for my mutation which takes a list of things as an argument. For example:
```
  mutation($tags: [tag_insert_input!]!) {
    insert_tag(objects: $tags) {
      affected_rows
      returning {
        id
      }
    }
  }
```

Stacktrace:
```
RangeError: Maximum call stack size exceeded
 at /Users/bday/Code/ts-graphql-plugin-reproduce/node_modules/ts-graphql-plugin/lib/typegen/type-gen-visitor.js:131:33
    at Array.forEach (<anonymous>)
    at visitVariableType (/Users/bday/Code/ts-graphql-plugin-reproduce/node_modules/ts-graphql-plugin/lib/typegen/type-gen-visitor.js:129:70)
    at /Users/bday/Code/ts-graphql-plugin-reproduce/node_modules/ts-graphql-plugin/lib/typegen/type-gen-visitor.js:131:33
    at Array.forEach (<anonymous>)
    at visitVariableType (/Users/bday/Code/ts-graphql-plugin-reproduce/node_modules/ts-graphql-plugin/lib/typegen/type-gen-visitor.js:129:70)
    at /Users/bday/Code/ts-graphql-plugin-reproduce/node_modules/ts-graphql-plugin/lib/typegen/type-gen-visitor.js:131:33
    at Array.forEach (<anonymous>)
    at visitVariableType (/Users/bday/Code/ts-graphql-plugin-reproduce/node_modules/ts-graphql-plugin/lib/typegen/type-gen-visitor.js:129:70)
    at /Users/bday/Code/ts-graphql-plugin-reproduce/node_modules/ts-graphql-plugin/lib/typegen/type-gen-visitor.js:131:33
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```
I created a project to reproduce the issue here: https://github.com/bday2009/ts-graphql-plugin-reproduce

This is a fairly common pattern and I guess everybody else is doing it, however I couldn't find a similar issue. Maybe it's something I overlooked on my part?

Thank you in advanced.

**Comment by Quramy** [comment]
I'll investigate this.

**Comment by Quramy** [comment]
@bday2009 

I fixed this bug and published ts-graphql-plugin v2.1.2.

And I confirmed the new version can generate types for your reproducing repository's query :  

```ts
/* eslint-disable */
/* This is an autogenerated file. Do not edit this file directly! */
type uuid_comparison_expInputType = {
    _eq?: any | null;
    _gt?: any | null;
    _gte?: any | null;
    _in?: any[] | null;
    _is_null?: boolean | null;
    _lt?: any | null;
    _lte?: any | null;
    _neq?: any | null;
    _nin?: any[] | null;
};
type String_comparison_expInputType = {
    _eq?: string | null;
    _gt?: string | null;
    _gte?: string | null;
    _ilike?: string | null;
    _in?: string[] | null;
    _is_null?: boolean | null;
    _like?: string | null;
    _lt?: string | null;
    _lte?: string | null;
    _neq?: string | null;
    _nilike?: string | null;
    _nin?: string[] | null;
    _nlike?: string | null;
    _nsimilar?: string | null;
    _similar?: string | null;
};
type timestamptz_comparison_expInputType = {
    _eq?: any | null;
    _gt?: any | null;
    _gte?: any | null;
    _in?: any[] | null;
    _is_null?: boolean | null;
    _lt?: any | null;
    _lte?: any | null;
    _neq?: any | null;
    _nin?: any[] | null;
};
type Boolean_comparison_expInputType = {
    _eq?: boolean | null;
    _gt?: boolean | null;
    _gte?: boolean | null;
    _in?: boolean[] | null;
    _is_null?: boolean | null;
    _lt?: boolean | null;
    _lte?: boolean | null;
    _neq?: boolean | null;
    _nin?: boolean[] | null;
};
type todo_bool_expInputType = {
    _and?: (todo_bool_expInputType | null)[] | null;
    _not?: todo_bool_expInputType | null;
    _or?: (todo_bool_expInputType | null)[] | null;
    created_at?: timestamptz_comparison_expInputType | null;
    id?: uuid_comparison_expInputType | null;
    is_completed?: Boolean_comparison_expInputType | null;
    tags?: tag_bool_expInputType | null;
    title?: String_comparison_expInputType | null;
    updated_at?: timestamptz_comparison_expInputType | null;
};
type tag_bool_expInputType = {
    _and?: (tag_bool_expInputType | null)[] | null;
    _not?: tag_bool_expInputType | null;
    _or?: (tag_bool_expInputType | null)[] | null;
    id?: uuid_comparison_expInputType | null;
    name?: String_comparison_expInputType | null;
    todo?: todo_bool_expInputType | null;
    todo_id?: uuid_comparison_expInputType | null;
};
type tag_on_conflictInputType = {
    constraint: "tag_pkey";
    update_columns: ("id" | "name" | "todo_id")[];
    where?: tag_bool_expInputType | null;
};
type tag_arr_rel_insert_inputInputType = {
    data: tag_insert_inputInputType[];
    on_conflict?: tag_on_conflictInputType | null;
};
type todo_insert_inputInputType = {
    created_at?: any | null;
    id?: any | null;
    is_completed?: boolean | null;
    tags?: tag_arr_rel_insert_inputInputType | null;
    title?: string | null;
    updated_at?: any | null;
};
type todo_on_conflictInputType = {
    constraint: "todo_pkey";
    update_columns: ("created_at" | "id" | "is_completed" | "title" | "updated_at")[];
    where?: todo_bool_expInputType | null;
};
type todo_obj_rel_insert_inputInputType = {
    data: todo_insert_inputInputType;
    on_conflict?: todo_on_conflictInputType | null;
};
type tag_insert_inputInputType = {
    id?: any | null;
    name?: string | null;
    todo?: todo_obj_rel_insert_inputInputType | null;
    todo_id?: any | null;
};
export type QueryResult = {
    insert_tag: ({
        affected_rows: number;
        returning: ({
            id: any;
        })[];
    }) | null;
};
export type QueryVariables = {
    tags: tag_insert_inputInputType[];
};
```

**Comment by bday2009** [comment]
Thank you very much!
